qiskit/quantum_info/synthesis/__init__.py

```diff
@@ -21,6 +21,10 @@
     TwoQubitBasisDecomposer,
     two_qubit_cnot_decompose,
 )
+from qiskit.quantum_info.quaternion import Quaternion
+
+from .cnotdihedral_decompose import decompose_cnotdihedral
+from .clifford_decompose import decompose_clifford
```

Comment:
I'm not sure that you need to add clifford_decompose and cnotdihedral_decompose, since they were deprecated in a previous release. 
Quaternion was moved inside quantum_info.

Reply:
This is true, but I'm restoring them here because 0.46.0 is special. We're documenting this release as being fully compatible with 0.45.x and it's primarily about just adding the deprecation warnings for API changes coming in 1.0.0.  Removing these as part of 0.46.0 would break that messaging because if someone was still using the deprecated `qiskit.quantum_info.synthesis.clifford_decompose()` function when they upgraded from 0.45.2 to 0.46.0 it would break. If 0.46.0 was a normal minor version release this would have been totally fine, 0.46.0 is really just a special edge case as the last minor version in the 0.* release series.